screens/Contacts/ExploreScreen.tsx

```diff
@@ -1,45 +1,118 @@
-import { useState } from "react" 
-import { View } from "react-native" 
-import { styles } from "./styles" 
+import React, { useEffect, useState } from "react"
+import { View } from "react-native"
+import { styles } from "./styles"
 import { useSafeAreaInsets } from "react-native-safe-area-context"
 import SectionTabs from "../../components/SectionTabs/SectionTabs"
 import { NavigationProp, ParamListBase } from "@react-navigation/native"
 import ContactList from "./ContactList/ContactList"
 import ContactGraph from "./ContactGraph/ContactGraph"
+import { mockContacts } from "./mockContacts"
+import Graph from "../../components/Graph/Graph"
+import Contact from "./Contact"
+import AsyncStorage from "@react-native-async-storage/async-storage"
 
-interface ContactListScreenProps{
+const GRAPH_STORAGE_KEY = "graph"
+const GRAPH_EXISTENCE_FLAG_KEY = "graph_exists"
+
+interface ContactListScreenProps {
   navigation: NavigationProp<ParamListBase>
 }
 
-const ExploreScreen = ({navigation} : ContactListScreenProps ) => {
+const ExploreScreen = ({ navigation }: ContactListScreenProps) => {
+  const [graph, setGraph] = useState<Graph>()
+
+  useEffect(() => {
+    const loadGraphData = async () => {
+      try {
+        // Check if the graph data exists in AsyncStorage
+        const graphData = await AsyncStorage.getItem(GRAPH_STORAGE_KEY)
+        if (graphData) {
+          // If the data exists, parse it and set the graph state
+          const parsedGraph = JSON.parse(graphData)
+          setGraph(parsedGraph)
+        } else {
+          // If the data doesn't exist, create the graph from contacts
+          const newGraph = createGraphfromContacts(mockContacts, "0")
+          // Store the graph data in AsyncStorage
+          await AsyncStorage.setItem(
+            GRAPH_STORAGE_KEY,
+            JSON.stringify(newGraph)
+          )
+
+          await AsyncStorage.setItem(GRAPH_EXISTENCE_FLAG_KEY, "true")
+          // Set the graph state
+          setGraph(newGraph)
+        }
+      } catch (error) {
+        console.error("Error loading graph data:", error)
+      }
+    }
```

Comment:
There you have a perfect example of where comments are needed - a function that contains many if statements and its' internal structure is not trivial to understand without comments